### ISSUE TITLE
JSON Modeling Techniques with dbt

### ISSUE THREAD

**Issue by wasilaq** [fishtown-analytics/docs.getdbt.com#1863]
### Contact Details

@Wasila

### I have read the dbt Developer Blog contribution guidelines.

- [X] I have read the dbt Developer Blog contribution guidelines.

### Which of these best describes you?

- [ ] I am a dbt Community member or partner contributing to the Developer Blog
- [X] I work for dbt Labs and am creating this issue for a community or marketing approved piece.

### What is the topic of your post?

This is a revival of [this Github discussion](https://github.com/dbt-labs/dbt-technical-blog-writing/discussions/75)

This would be a blog series composed of three parts (introduction to JSON, working with JSON, and modeling JSON in dbt). It'll walk through an introduction of semi-structured data (why is it useful and where might you run into it?), how JSON is formatted, and how you can work with JSON in Snowflake. The final post is a review of different JSON modeling techniques and the pros and cons of each option.

This post is targeted towards intermediate to advanced dbt users who have JSON objects in their warehouse and are curious about best practices on how to transform.

### Link to an initial outline.

[Outline Notion doc](https://www.notion.so/dbtlabs/JSON-Article-Planning-ea488f97efff4474b95bd4a704809786)

A draft for part 3 has been written out by @christineberger and you can find it [here](https://www.notion.so/dbtlabs/WIP-JSON-Modeling-Techniques-dbt-acba1abb777d41938fd4c0317f8b1b6c). Parts 1 and 2 still need to be written.

**Comment by KiraFuruichi** [comment]
Hey @wasilaq thank you for submitting this issue, we are super pumped to hear y'all want to write about this!!

We were thinking about how this sort of material could best work for practitioners and came up with the following idea, please let us know how you feel about it!
- Part 1 being a JSON glossary page in the [Analytics Engineering Glossary ](https://docs.getdbt.com/glossary)(you'd be able work with @johnblust on structuring that in a way that works for y'all)
- Part 2 & 3 we think would make an excellent best practices guide on working with JSON data that lives on the [Guides section](https://docs.getdbt.com/guides/best-practices) on the docs site. @jasnonaz and I would work on editing that with you and Christine.

How does that sit for you two?

**Comment by wasilaq** [comment]
Thanks @KiraFuruichi! Sounds good to me. I'll work on the glossary page and aim to get a rough draft to @johnblust sometime this week!

**Comment by KiraFuruichi** [comment]
Awesome!! Feel free to reference other [Glossary posts](https://docs.getdbt.com/terms/surrogate-key) to see the general structure used, but @johnblust will be able to provide you with some outline pointers too!

**Comment by johnblust** [comment]
Hey @wasilaq, here's a content brief for the [JSON glossary page](https://www.notion.so/dbtlabs/JSON-d3eb2888afb344adb31c7f5be39514a8). Let me know if you have any questions, if you like there's a section we should take out, or if you feel like anything is missing. Otherwise, feel free to get started on an outline :) When the outline is done, you can share it with me and I'll review it. Once the outline is approved, we can move on to the draft stage!

@KiraFuruichi, should Wasila create an Issue for this piece, or should we keep everything in this thread?

**Comment by KiraFuruichi** [comment]
Good question--let's leave it in one issue for now and if it gets long, we can create a separate issue for the guide!

**Comment by wasilaq** [comment]
@johnblust Just shared [a Google doc](https://docs.google.com/document/d/1GMCw9YKODX3NH5bYX224cyRdbJa0PZZy07EhDQkTNsA/edit?usp=sharing) with you! Feel free to add feedback whenever you can and let me know if there's anything you want me to add to the outline.

Sorry for the delayed responses y'all, I haven't been getting notifications for this and wasn't able to figure out why but I think I fixed the issue this morning.

**Comment by johnblust** [comment]
@wasilaq This is a great start! I left some comments, mainly around the examples you have planned. If you could expound on what those examples will look like, that would be great!

**Comment by wasilaq** [comment]
> @wasilaq This is a great start! I left some comments, mainly around the examples you have planned. If you could expound on what those examples will look like, that would be great!

Done :)

**Comment by johnblust** [comment]
Awesome! Feel free to get started on the draft whenever you're ready

**Comment by wasilaq** [comment]
Finished the [first draft](https://docs.google.com/document/d/1xljsKTXU3WhREgrR-PlW6Snik0aY-sHI0szW0GEEz5Y/edit?usp=sharing). Would love some feedback. I'm thinking that the last few sections might need some fleshing out.

**Comment by johnblust** [comment]
@wasilaq This looks good from my perspective. @KiraFuruichi if you could give this a once over from a technical perspective, I would really appreciate it! Once that is done, I think you're good to spin up a PR for this!

**Comment by wasilaq** [comment]
Awesome! I'll also see if anyone on the proserv team has feedback. I know we have folks there who have more experience with JSON so it would be nice to get their eyes on it.

**Comment by johnblust** [comment]
Yes, that could work as a technical review as well!

**Comment by wasilaq** [comment]
Draft has been cleaned up! Should I open a PR for this?

**Comment by johnblust** [comment]
Awesome!! Usually yes, but I'll spin up a PR for this one. I need to play around with a couple of different options to see if we can get dropdown functionality for those questions you have listed. If I can't get it to work, I'll lyk here and we can figure out what to do about it.

**Comment by johnblust** [comment]
@wasilaq @christineberger Hey Wasila and Christine! Congrats on the glossary entry release!! Really appreciate the hard work there.

As for creating the guide, you all can get started working on that and mapping out what that could look like. The DevHub team is pretty heads down on Coalesce stuff, so if you need additional guidance on that, we won't have the bandwidth to help until after Coalesce. Thanks again for your help on this! Looking forward to seeing what you two come up with :)

**Comment by KiraFuruichi** [comment]
Hey @wasilaq! Just added the "Guide" tag to this issue as y'all closed out the glossary page!

**Comment by wasilaq** [comment]
Posting here to hold myself accountable- hoping to have some sort of draft together by the end of next week! Ideally something I can bring to the devhub writing session on Thursday.

**Comment by wasilaq** [comment]
Don't anticipate being able to get back to this for a while, so posting the current state of the draft [here](https://docs.google.com/document/d/1Yv5cMXsA1dQEmWDmTDc0KPIvHXeKGgYoZ6JrDcTf1Rc/edit). Anyone's free to pick it up in the mean time, but not a priority

**Comment by runleonarun** [comment]
I marked this as blocked because it needs some reviews, and I believe it should be a dev blog and not a guide.

**Comment by mirnawong1** [comment]
hey team, going to close this issue since it's still ongoing and blocked. when we're ready, please feel free to re-open i up again when its ready for publishing!
